File: rb109/Practice/Easy 5/bannerizer.rb
# bannerizer.rb


def print_in_box(message)
  puts "+" + "-"*(message.length + 2) + "+"
  puts "|" + ' '*(message.length + 2) + "|"
  puts "| " + message + " |"
  puts "|" + ' '*(message.length + 2) + "|"
  puts "+" + "-"*(message.length + 2) + "+"
end

print_in_box('To boldly go where no one has gone before.')
print_in_box('Matt Johnston')
print_in_box('Not as painful as I was expecting. Phew.')
print_in_box('')

File: js210/js211_assessment/js210sp_easy2/bannerizer.js
"use strict";

function logInBox(text) {
  let width = text.length + 2;
  let border = '+' + borderMaker('-', width) + '+';
  let margin = '|' + borderMaker(' ', width) + '|';
  let textLine = '| ' + text + ' |';

  console.log(border);
  console.log(margin);
  console.log(textLine);
  console.log(margin);
  console.log(border);
}


function borderMaker(char, length) {
  let result = '';

  while (result.length < length) {
    result += char;
  }

  return result;
}

logInBox('To boldly go where no one has gone before.');
logInBox('');
